src/services/validateZendeskRequest.ts

```diff
@@ -1,17 +1,158 @@
 import { ValidatedDataRequestParamsResult } from '../types/validatedDataRequestParamsResult'
+import { getEpochDate, tryParseJSON } from '../utils/helpers'
+
+function isEmpty(obj: object): boolean {
+  return obj && Object.keys(obj).length === 0
+}
+
+const IDENTIFIERS = ['event_id', 'session_id', 'journey_id']
+
+const VALID_PII_TYPES = [
+  'passport_number',
+  'passport_expiry_date',
+  'drivers_license',
+  'name',
+  'dob',
+  'current_address',
+  'previous_address'
+]
 
 export const validateZendeskRequest = (
   body: string | null
 ): ValidatedDataRequestParamsResult => {
-  console.log(body)
+  const data = tryParseJSON(body ?? '{}')
+  if (isEmpty(data)) {
+    return {
+      validationMessage: 'No data in request',
+      isValid: false
+    }
+  }
+  const isEmailValid = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*\.gov.uk$/.test(
+    data.resultsEmail ?? ''
+  )
+
+  const piiTypes = data.piiTypes.replace(/,/g, '')
+  const piiTypesValidated = !piiTypes.length || /[^,(?! )]+/gm.test(piiTypes)
+  const piiTypesList = mapSpaceSeparatedStringToList(data.piiTypes)
+  const piiTypesAllValid = piiTypesList?.length
+    ? piiTypesList.every((type) => VALID_PII_TYPES.includes(type))
+    : true
+  const fieldValidation = [
+    {
+      message: 'Email format invalid',
+      isValid: isEmailValid
+    },
+    {
+      message: 'At least one session id should be provided',
+      isValid:
+        data.identifierType != 'session_id' || data.sessionIds?.length > 0
+    },
+    {
+      message: 'At least one journey id should be provided',
+      isValid:
+        data.identifierType != 'journey_id' || data.journeyIds?.length > 0
+    },
+    {
+      message: 'At least one event id should be provided',
+      isValid: data.identifierType != 'event_id' || data.eventIds?.length > 0
+    },
```

Comment:
There is the  `User ID` identifier also. Should we be checking for that as well?